test/activation_based/test_learning.py

```
@@ -0,0 +1,180 @@
+    return learners
+
+
+def test_stdp_learner_records_are_detached():
+    net = _build_net("m")
+    learners = _build_learners(net, "m")
+
+    x = (torch.rand(4, 2, 1, 8, 8) > 0.5).float()
+    y = net(x)
+
+    assert y.requires_grad
+    for l in learners:
```

Ruff (0.15.20)

[error] Ambiguous variable name: `l`

(E741)